src/xtgeo/grid3d/grid_property.py

```diff
@@ -465,127 +580,133 @@ def _set_initial_dimensions(self, gridlike, input_dimensions):
+
         Raises:
-            ValueError: if given dimensions are not None and do not
+            ValueError
+                If given dimensions are not None and do not
                 match dimensions of the GridProperty
+
         """
         if ncol is not None and self._ncol != ncol:
             raise ValueError(
-                f"mismatching column dimension given: {ncol} vs {self._ncol}"
+                f"Mismatching column dimension given: {ncol} vs {self._ncol}"
             )
         if nrow is not None and self._nrow != nrow:
-            raise ValueError(f"mismatching row dimension given: {nrow} vs {self._nrow}")
+            raise ValueError(f"Mismatching row dimension given: {nrow} vs {self._nrow}")
         if nlay is not None and self._nlay != nlay:
             raise ValueError(
-                f"mismatching layer dimension given: {nlay} vs {self._nlay}"
+                f"Mismatching layer dimension given: {nlay} vs {self._nlay}"
             )
 
-    def __del__(self):
+    def __del__(self) -> None:
         logger.debug("DELETING property instance %s", self.name)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         myrp = (
             f"{self.__class__.__name__} (id={id(self)}) ncol={self._ncol!r}, "
             f"nrow={self._nrow!r}, nlay={self._nlay!r}, filesrc={self._filesrc!r}"
         )
         return myrp
 
-    def __str__(self):
-        # user friendly print
+    def __str__(self) -> str:
         return self.describe(flush=False)
 
     # ==================================================================================
     # Properties
-    # Some proprerties such as ncol, nrow, nlay are from the Super class
+    # Some propreties such as ncol, nrow, nlay are from _Grid3d
     # ==================================================================================
 
     @property
-    def metadata(self):
-        """Return metadata object instance of type MetaDataRegularSurface."""
+    def metadata(self) -> MetaDataCPProperty:
+        """Get or set metadata object instance of type MetaDataCPProperty."""
         return self._metadata
 
     @metadata.setter
-    def metadata(self, obj):
-        # The current metadata object can be replaced. A bit dangerous so further
-        # check must be done to validate. TODO.
-        if not isinstance(obj, xtgeo.MetaDataCPProperty):
-            raise ValueError("Input obj not an instance of MetaDataCPProperty")
-
-        self._metadata = obj  # checking is currently missing! TODO
+    def metadata(self, metadata: MetaDataCPProperty) -> None:
+        if not isinstance(metadata, MetaDataCPProperty):
+            raise ValueError("Input metadata not an instance of MetaDataCPProperty")
+        # TODO: validate this?
+        self._metadata = metadata
 
     @property
-    def name(self):
-        """Returns or rename the property name."""
+    def name(self) -> Optional[str]:
+        """Get or set the property name."""
```

Comment:
Make `Get or set` consistently prepended in documentation for properties with getters+setters